src/ja/common/job.py

```diff
@@ -0,0 +1,103 @@
+'''
+This module contains the definition of the Job class and related structures.
+'''
+from enum import Enum
+from docker_context import DockerConstraints, IDockerContext
+
+
+class JobStatus(Enum):
+    '''
+    Represents the status of a job.
+    '''
+    NEW = 0
+    QUEUED = 1
+    RUNNING = 2
+    PAUSED = 3
+    DONE = 4
+    KILLED = 5
+    CRASHED = 6
+
+
+class JobPriority(Enum):
+    '''
+    Represents the priority of a job.
+    '''
+    LOW = 0
+    MEDIUM = 1
+    HIGH = 2
+    URGENT = 3
+
+
+class Job:
+    '''
+    Represents a job with all of its attributes.
+    '''
+    def __init__(self,
+                 owner_id: int,
+                 priority: JobPriority,
+                 ctx: IDockerContext,
+                 constraints: DockerConstraints):
+        self._uid = None
+        self._status = JobStatus.NEW
+        self._priority = priority
+        self._ownerid = owner_id
+        self._ctx = ctx
+        self._constraints = constraints
+
+    @property
+    def uid(self) -> str:
```

Comment:
Should this be str or int?
If a user later queries the server is this going to be a visible property?

Reply:
In our requirements we defined UID as a string consisting of letters, digits and -
And yes it will be visible, otherwise the user can't say which of their jobs to cancel (or which job to query information about)

Reply:
I see.
It seems I got confused.
Should UID be part of the constructor?

Reply:
I was initially thinking it should be in the constructor, but after all, we can't get a UID for the job in the user client - the server assigns UIDs (otherwise we *might* have collisions). So a job initially has no UID, and when it was Queued, the server should respond back with a UID.

However, we may want to take a different approach. I am also not sure where this piece of information belongs. Should I add it in the comment here?

Reply:
I think it should work like this:

* UID is None by default. In this case the server automatically assigns a UID when it receives the Job.
* A user can specify a desired UID when creating the Job.
* If the server detects a collision it renames the Job (something like Tensorflow -> Tensorflow-2).